lib/rack/attack/configuration.rb

```diff
@@ -13,11 +13,18 @@ def initialize
         @tracks = {}
         @anonymous_blocklists = []
         @anonymous_safelists = []
+        @throttled_response_retry_after_header = false
 
         @blocklisted_response = lambda { |_env| [403, { 'Content-Type' => 'text/plain' }, ["Forbidden\n"]] }
         @throttled_response   = lambda do |env|
-          retry_after = (env['rack.attack.match_data'] || {})[:period]
-          [429, { 'Content-Type' => 'text/plain', 'Retry-After' => retry_after.to_s }, ["Retry later\n"]]
+          if throttled_response_retry_after_header
+            match_data = env['rack.attack.match_data']
+            now = match_data[:epoch_time]
+            retry_after = match_data[:period] - (now % match_data[:period])
```

Comment:
hi, don't we need `+ 1` here to match [expired_in](https://github.com/kickstarter/rack-attack/blob/master/lib/rack/attack/cache.rb#L65)?

Reply:
Seems like we do...
Thanks @sonalkr132 

@fatkodima Any thoughts about this?

Reply:
I can't say for 100% sure right now, but seems like we don't need that +1.
In `expired_in` that +1 is used to set key expiration key, but here `retry_after` describes when the next period will begin to start accepting requests.
@sonalkr132 can you please investigate more, if it looks like I'm not correct.

Reply:
yes, you are right. acceptance of new requests depends on the period window and not key expiry time. Thank you for explaining,